### PR TITLE
Represent EDITNNC in the deck representation.

### PR DESCRIPTION
**Pull request by blattms**
This PR adds EDITNNC to the deck to be queried downstream. As NNCs can come from various sources and might be computed by the simulator we cannot edit them in the parser. Therefore we now add a member on the EcipseState to represent this information. The simulator can query and apply it.

This reverts the previous try in PR #535.

### CODE REVIEW

File: opm/parser/eclipse/EclipseState/Edit/EDITNNC.hpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    /// Construct from input deck
+    explicit EDITNNC(const Deck& deck);
+    /// \brief Get an ordered set of EDITNNC
+    const std::vector<NNCdata>& data() const
+    {
+        return m_editnnc;
+    }
+    /// \brief Get the number of entries
+    size_t size() const;
+    /// \brief Whether EDITNNC was empty.
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Edit/EDITNNC.hpp]: I know opinions go in all directions on this - but personally I consider comments of the type: `Get the number of entries` on a method called `size()` to be just noise.

File: src/opm/parser/eclipse/EclipseState/Edit/EDITNNC.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+#include <opm/parser/eclipse/EclipseState/Grid/GridDims.hpp>
+#include <opm/parser/eclipse/EclipseState/Edit/EDITNNC.hpp>
+#include <opm/parser/eclipse/Parser/ParserKeywords/E.hpp>
+#include <opm/common/OpmLog/OpmLog.hpp>
+
+
+namespace Opm
+{
+
+void readEditNncs(const std::vector< const DeckKeyword* >& editNncsKw, std::vector<NNCdata>& editNncs, const GridDims& gridDims)
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/EclipseState/Edit/EDITNNC.cpp]: I guess this could be in a anonymous namespace?

**Comment by blattms** [comment]
Sure it could be moved into an anonymous namespace. What would be the
benefit?
Could also be a private class function of course.

File: opm/parser/eclipse/EclipseState/Edit/EDITNNC.hpp (hunk 1 above)
Comment (joakim-hove) [reply opm/parser/eclipse/EclipseState/Edit/EDITNNC.hpp]: I would add `begin()` and `end() ` methods to thus class.

**Comment by joakim-hove** [comment]
I have some minor comments, but they are just suggestions. Feel free to merge this when you are ready -  or give me a hint and I will do it.

**Comment by blattms** [comment]
rebased onto master.

File: opm/parser/eclipse/EclipseState/Edit/EDITNNC.hpp (hunk 1 above)
Comment (blattms) [reply opm/parser/eclipse/EclipseState/Edit/EDITNNC.hpp]: I took NNC as a template for this class. That one must missing that, too. I am not opposed to adding them but maybe we should do this consistently for both classes in a separate PR.

**Comment by blattms** [comment]
Will do once I got feedback of tests from other people that EDITNNC
now really works.

**Comment by blattms** [comment]
jenkins build this ewoms=425 please

**Comment by blattms** [comment]
In @joakim-hove's spirit I added a test to this and immediately discovered a bug that is fixed now (obscured by a fixup and forced push). Removing WIP as this is tested.

**Pull request merged**